tools/text_processing/split_file_on_column/split_file_on_column.xml

```diff
@@ -1,17 +1,41 @@
-<tool id="tp_split_on_column" name="Split file" version="0.3">
+<tool id="tp_split_on_column" name="Split file" version="0.2">
     <description>according to the values of a column</description>
     <requirements>
-        <requirement type="package" version="4.1.3">gawk</requirement>
+        <requirement type="package" version="4.1.0">gnu_awk</requirement>
     </requirements>
     <command>
 <![CDATA[
         mkdir tmp_out &&
-        awk -F'\t' '{print > "tmp_out/"\$$column".$infile.ext" }' '$infile'
+
+	#if $condition_has_header.has_header =="y"
+	   #if $condition_has_header.include_header =="y"
+	      awk -F '\t' 'NR==1{hdr=$0;next}f!="tmp_out/"\$$column".$infile.ext"{if(f) close(f); f="tmp_out/"\$$column".$infile.ext";print hdr>f} {print >> f}' $infile
+	   #else
+              awk -F '\t' 'NR==1{hdr=$0;next}f!="tmp_out/"\$$column".$infile.ext"{if(f) close(f); f="tmp_out/"\$$column".$infile.ext"} {print >> f}' $infile
+	   #end if
+        #else
+           awk -F'\t' '{print > "tmp_out/"\$$column".$infile.ext" }' '$infile'
+        #end if
 ]]>
     </command>
     <inputs>
         <param format="tabular" name="infile" type="data" label="File to select" />
         <param name="column" label="on column" type="data_column" data_ref="infile" accept_default="true" />
+        <conditional name="condition_has_header">
+            <param name="has_header" type="select" display="radio" label="Does the input file has header?">
+                <option value="n">No</option>
+                <option value="y">Yes</option>
+            </param>
+            <when value="y">
+                <param name="include_header" type="select" display="radio" label="Include header in all splitted files?" value="n">
```

Comment:
why is the conditional needed? This param is enough isn't it?

Reply:
i'm just thinking users could upload a file with header but do not want to include header in the splitted files.
or upload a file with header and also include header in the split files.

if user upload a file without header, it functions as before...